src/System.Text.Encoding.CodePages/src/System/Text/SBCSCodePageEncoding.cs

```diff
@@ -821,8 +821,7 @@ public override unsafe int GetCharCount(byte* bytes, int count, DecoderNLS decod
             byte[] byteBuffer = new byte[1];
             char* charEnd = chars + charCount;
 
-            DecoderFallbackBufferHelper fallbackHelper = new DecoderFallbackBufferHelper(
-                decoder != null ? decoder.FallbackBuffer : DecoderFallback.CreateFallbackBuffer());
+            DecoderFallbackBufferHelper fallbackHelper = new DecoderFallbackBufferHelper(fallbackBuffer);
```

Comment:
I think it'd be clearer if it were just `new DecoderFallbackBufferHelper(null)`.

Reply:
I can use `null` instead, I just did it this way because that's how it's done in `DBCSCodePageEncoding`.

Reply:
Right, you can use DecoderFallbackBufferHelper fallbackHelper = null here as we always we check for null later.

Reply:
> DecoderFallbackBufferHelper fallbackHelper = null

DecoderFallbackBufferHelper is a struct.